### ISSUE TITLE
No video from X1C or P1S, yet Bambu Slicer no problem.

### ISSUE THREAD

**Issue by tlhintoq** [SoftFever/OrcaSlicer#5170]
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### OrcaSlicer Version

2.0 nightly since 25apr2024

### Operating System (OS)

Windows

### OS Version

Windows10x64

### Printer

Bambu X1C or P1S

### How to reproduce

From Device tab, try to see streaming video from either printer

### Actual results

![image](https://github.com/SoftFever/OrcaSlicer/assets/11351506/44094b7c-0547-41f8-97b8-6a109e833395)

As you can see, Bambu slicer can connect.  
Log out then close down Bambu slicer.  Start Orca then log in to Bambu network.  Nope, no video - but will print to it just fine.

### Expected results

If Bambu Slicer and see the video feeds then so should Orca Slicer, right?

### Project file & Debug log uploads

Let me know what you need.

### Checklist of files to include

- [X] Log file
- [X] Project file

### Anything else?

Maybe a Bambu network plugin that didn't update with the release version of Orca 2?
What do you need me to look at to confirm versions?

**Comment by KD7LRJ** [comment]
Same for me. Handy works on my Android phone, but 2.1.0-dev gives a black screen with an error message, "Please check the network and try again. You can restart or update the printer if the issue persists. [2:-10]"

Same results for my 4 P1P and 2 X1C machines.

Un/reinstall hasn't helped. Send/Printing from my Win 11 desktop works fine, just not the video feed.

**Comment by tlhintoq** [comment]
Hey Douglas, Thanks for the confirmation that it wasn't just me - I mean I'm sorry its happening to you, but its good to know its more widespread than just one bad installation.

73's
KC9RY - Clint

**Comment by tlhintoq** [comment]
Just downloaded 2.1dev 28apr.
Uninstalled old version.
Ran installer as admin.
No flags, errors, warnings, issues.

Ran the newly installed Orca.  I was hopeful when I saw it downloading the Bambu network plug-in, thinking that might be latest everything.
![2024-04-29_17-50-26](https://github.com/SoftFever/OrcaSlicer/assets/11351506/3a3eb4ad-aa79-4730-bbcf-0a0fdf7bf99f)

No joy.  Still able to print via network to X1C and P1S - but no video stream from either.
FYI: I have a static IP to my house, so its not NAT or some other broadband issue.
![2024-04-29_17-50-47](https://github.com/SoftFever/OrcaSlicer/assets/11351506/e3715e2f-3ae4-45c9-8990-7ef3b5a4e061)

**Comment by TexZeTech** [comment]
![image](https://github.com/SoftFever/OrcaSlicer/assets/7254540/6671d228-7fb0-4afe-b327-737104cfe134)

@tlhintoq You didn't upload any log files or project files...... kinda hard to look at what may or may not be happening with out those. So to start upload your log files.

**Comment by tlhintoq** [comment]
![2024-05-01_13-46-06](https://github.com/SoftFever/OrcaSlicer/assets/11351506/c84a47ff-6701-419d-abd8-9127ecab5d54)
[debug_network_Wed_May_01_13_44_16.log.zip](https://github.com/SoftFever/OrcaSlicer/files/15173230/debug_network_Wed_May_01_13_44_16.log.zip)

**Comment by pgifford** [comment]
This should be easily reproducible.  The error code (from [here](https://wiki.bambulab.com/en/software/bambu-studio/faq/live-view)) is: 

> <html><body>

-10 | Certificate error | Check if you are using the official version of Bambu Studio
-- | -- | --

</body>
</html>

As noted Live View works fine in Bambu Studio, also works in Bambu Handy.

**Comment by tlhintoq** [comment]
X1C got latest firmware yesterday.
Today OrcaSlicer just prompted for new network plug in.  
This seemed like a match and maybe would resolve the issue.
No joy.  Still fails with the `2:-10` error

**Comment by tlhintoq** [comment]
10may2024
Grabbed the latest overnight.
Ran installer as admin
Let it uninstall previous version.
Installed new version.
Run it
Prompt for new network plugin - `[OK]`
Close OrcaSlicer (because it takes effect on the next launch)
![2024-05-10_16-28-10](https://github.com/SoftFever/OrcaSlicer/assets/11351506/e7a3b360-3b7b-4797-adf3-437e262d419b)

Launch again.
Log out of Bambu network
Log in to Bambu network
power cycle the X1C and the P1S

All other network functions are good.  But no video stream.  BambuStudio has no problem with video stream so its not the network, PC or printers.
![image](https://github.com/SoftFever/OrcaSlicer/assets/11351506/d343b29c-f64c-422f-a889-2e619d920233)

**Comment by tlhintoq** [comment]
New network plugin alert.
![2024-05-12_16-31-52](https://github.com/SoftFever/OrcaSlicer/assets/11351506/47e556f4-3b58-4473-93a9-9aa654210210)
Same proper steps as before including reboots etc.
No joy.  Still getting no video with `2:-10` error.

**Comment by pgifford** [comment]
Video is working for me now after the latest network plugin installed.

**Comment by tlhintoq** [comment]
> Video is working for me now after the latest network plugin installed.

What is "latest" for you?  "01.09.01.06" like in my screen shot, or newer?
Which firmware on your machine?  X1C or P1S ?
Which Orca?  2.1.0-dev?  Nightly build?

I'm running dev from 2 days ago.
![image](https://github.com/SoftFever/OrcaSlicer/assets/11351506/3780fa31-bd68-4480-a90a-6c00b5dee095)

I just opened BambuStudio to deliberatley log in then log out.  So I'm not connected there.
Launched OrcaSlicer - log in then log out.  Not connected there.
Powered off both printers.
Powered off network router.
Basically rebooting the entire shop.
Power on the router
Powered on the printers.
Launch Orcaslicer - still get `2:-10` error on both X1C and P1S.
Close Orca, launch Bambu - video on both.
Close Bambu, launch Orca - both still fail.

![image](https://github.com/SoftFever/OrcaSlicer/assets/11351506/389d69d6-3def-4672-93a0-fdb407bce518)

![image](https://github.com/SoftFever/OrcaSlicer/assets/11351506/6441ef4c-97ef-4de2-89b9-d7300c574dbf)

**Comment by pgifford** [comment]
I'm using Orca Slicer v2.1.0-dev from 30 Apr 2024.  I don't know how to find the version of the network plugin being used.  When the "new network plugin" dialog pops up I always click OK. 

I just downloaded and ran the latest nightly Windows portable, video works fine.

A few days ago I installed Bambu Studio Beta for an unrelated reason, I don't know how or why that would affect Orca Slicer though but the timing is about right.  One nice thing about Bambu Studio is the Help | About box shows the plugin version.

<img width="423" alt="studio" src="https://github.com/SoftFever/OrcaSlicer/assets/6106848/95894f85-cb75-4c4f-b8df-dd479102f6db">

**Comment by tlhintoq** [comment]
>  just downloaded and ran the latest nightly Windows portable, video works fine.

Hmmm... "Portable version", huh?  I might have to try that and see if there is a difference to the full installed version.

**Comment by tlhintoq** [comment]
>  I might have to try that and see if there is a difference to the full installed version.

No joy.  Same error.

**Comment by pgifford** [comment]
> > I might have to try that and see if there is a difference to the full installed version.
> 
> No joy. Same error.

Try installing Bambu Studio Beta. I can’t say it fixed the problem but I can’t say it didn’t either.

**Comment by tlhintoq** [comment]
> > > I might have to try that and see if there is a difference to the full installed version.
> > 
> > 
> > No joy. Same error.
> 
> Try installing Bambu Studio Beta. I can’t say it fixed the problem but I can’t say it didn’t either.

What the heck...
Bambu beta installed.  It grabbed a network plug in.  It sees my printers.
![2024-05-13_16-38-07](https://github.com/SoftFever/OrcaSlicer/assets/11351506/145576e1-4241-468f-aa06-d2b9b4a5df7f)

Now to Orca
![2024-05-13_16-40-19](https://github.com/SoftFever/OrcaSlicer/assets/11351506/b897c8f1-55a7-45c1-b239-92c3bc6f4e91)


Holy crow - @pgifford you're a life saver.  Looks like the plug in from the beta is the key.  Must be a shared component between both applications.  Keeping in mind, updates to Orca could overwrite it.  _MIGHT_ have to re-install Bambu beta after any Orca updates if the video issue returns.

**Comment by KD7LRJ** [comment]
Confirmed! I hadn't updated Bambu Studio since Version 1.8 and the update process installed 1.9.1.66 (not beta). Once I started the app after installing, it also updated the network plugin. Orca Slicer is now working again to view video from my printers.

**Comment by mydevpeeps** [comment]
confirmed as well. worked fine on 2.1.0.

**Comment by pwr22** [comment]
I had the same issue after updating to Orca 2.1.0 beta and resolved it now by letting Bambu update. Thanks for the help folks 👍!

**Comment by brandonkish** [comment]
Using Orca Slicer 2.1.0 kept getting error "[2-10]" in Orca Slicer on 6/19/24 and before.

Updated to Orca Slicer 2.1.0-rc on 6/19/24. Continued to get error "[2-10]"

Confirmed currently installed Bambu Lab version is V1.8.2.56. Can see the camera's video in Bambu Lab V1.8.2.56. 

My Log files with Bambu Lab to V1.8.2.56 installed and Orca Studio V2.1.0-rc installed. With Orca Studio V2.1.0-rc still giving "[2-10]" error.
[log.zip](https://github.com/user-attachments/files/15895504/log.zip)

Downloaded and installed Bambu Lab 01.09.02.57.

Opened Bambu Lab V01.09.02.57. Confirmed I can see the camera video. Closed Bambu Lab V01.09.02.57. (Note: No notification that the "Network Plugin" need to be updated with this version after install)

Opened Orca Slicer V2.1.0-rc. Can now see Video on Device tab.

This is the logs for Orca Slicer V2.1.0-rc after downloading Bambu Lab V1.09.02.57 and being able to successfully see the video on the device tab.
[Orca Slicer Log After Installing Bambu Lab V1.09.02.57.zip](https://github.com/user-attachments/files/15895593/Orca.Slicer.Log.After.Installing.Bambu.Lab.V1.09.02.57.zip)


This is probably unrelated, but I got this warning message constantly when switching between the programs and after updating each program:
![image](https://github.com/SoftFever/OrcaSlicer/assets/7042606/ce27f4f3-dd28-4018-b19c-8c6106fbb673)
In bot Orca Slicer and Bambu Lab when I updated Orca to 2.1.0-rc, and Bambu Lab to V1.8.2.56. Not sure if this is related, but I hit "Yes" both times. Which brought up this window:
![image](https://github.com/SoftFever/OrcaSlicer/assets/7042606/05450f97-de01-4b52-a3f0-974407adfdd8)
Clicking "Yes" to this prompt did not seem to allow the video to display in Orca before updating Bambu Lab.

**Comment by brandonkish** [comment]
I also should note, at one point, when I installed Orca Slicer 2.1.0, Orca slicer gave me some kind of message that it could not load my printer configurations. This happened a month ago or so. The exact date is not clear to me.

After a brief moment of panicking, I restarted Orca Slicer, and all my profiles were sill there. I don't remember what I did to bring them back, or if I did anything. I am not sure if this is related, and which I could give you more details, but I am not sure if this had something to do with it, or if others also had this issue.

**Comment by mydevpeeps** [comment]
still an issue in GA 2.1.1 see https://github.com/SoftFever/OrcaSlicer/issues/5754#issuecomment-2209553656 for workaround.

**Comment by drizztguen77** [comment]
I was receiving the exact same warning about BambuSource from a different install. I tried the 2.10 version and then the 2.0 version. The 2.10 version still received the warning and then 2.0 version works fine. So there is something new in the 2.10 version that is causing an issue with the network driver. If I say "No" to not fix it, it still seems to work and shows me the camera.

**Comment by mlimseth** [comment]
Just install latest Bambu studio and the video come back on Orca.

**Comment by drizztguen77** [comment]
I have the latest version of Bambu Studio installed. I installed Orca 2.1.1 alongside 2.0.0. I get no warning at all when I use 2.0.0. When I use 2.1.1 I get the warning. This is with Bambu Studio 1.9.3.50 installed. I can answer No and it seems to work but it is still concerning to get the warning.

![2024-08-20 13_57_41-Untitled - BambuStudio](https://github.com/user-attachments/assets/cc81c652-06eb-47c8-a16f-4925d622235e)
![2024-08-20 14_10_09-Untitled - OrcaSlicer](https://github.com/user-attachments/assets/1151c185-c112-435a-8ce6-2be72181c8fa)

**Comment by motionbug** [comment]
Having the camera issue with Orca slicer version 2.3.0 video doesn't work but with Bambu studio 02.01.00.59 it does work

@ifihaffto I have the same issue with tyring to download the virtual camera tools, it downloads then tries to download again and we get into a loop, doesn't install with Bambu, get an error.
